pkg/route/api/validation/validation.go

```diff
@@ -76,6 +79,64 @@ func ValidateRouteStatusUpdate(route *routeapi.Route, older *routeapi.Route) fie
 	return allErrs
 }
 
+// ValidateRouteTLSConfig checks if the TLS config (certificates and keys) for a route is valid.
+func ValidateRouteTLSConfig(route *routeapi.Route) field.ErrorList {
+	tls := route.Spec.TLS
+	result := field.ErrorList{}
+
+	tlsFieldPath := field.NewPath("spec").Child("tls")
+	if errs := validateTLS(route, tlsFieldPath); len(errs) != 0 {
+		result = append(result, errs...)
+	}
+
+	// TODO: Check if we can be stricter with validating the certificate
+	//       is for the route hostname. Don't want existing routes to
+	//       break, so disable the hostname validation for now.
+	// hostname := route.Spec.Host
+	hostname := ""
+	certPool := x509.NewCertPool()
+
+	if len(tls.CACertificate) > 0 {
+		cacert, err := validateCertificatePEM(tls.CACertificate, nil)
+		if err != nil {
+			result = append(result, field.Invalid(tlsFieldPath.Child("caCertificate"), tls.CACertificate, err.Error()))
+		} else {
+			certPool.AddCert(cacert)
+		}
+	}
+
+	verifyOptions := &x509.VerifyOptions{
+		DNSName: hostname,
+		Roots:   certPool,
+	}
+
+	if len(tls.Certificate) > 0 {
+		if _, err := validateCertificatePEM(tls.Certificate, verifyOptions); err != nil {
+			result = append(result, field.Invalid(tlsFieldPath.Child("certificate"), tls.Certificate, err.Error()))
+		}
+	}
+
+	if len(tls.Key) > 0 {
+		if len(tls.Certificate) > 0 {
+			if err := validatePrivateKeyForCert(tls.Key, tls.Certificate); err != nil {
+				result = append(result, field.Invalid(tlsFieldPath.Child("key"), tls.Key, err.Error()))
+			}
+		} else {
+			if err := validatePrivateKeyPEM(tls.Key); err != nil {
```

Comment:
Well, today its really possible to specify both in one string - multiple PEM blocks in `tls.Certificate` and it would still be valid since the resulting PEM file contains required bits of information.